executor/joiner.go

```diff
@@ -381,6 +537,28 @@ func (j *leftOuterJoiner) tryToMatch(outer chunk.Row, inners chunk.Iterator, chk
 	return matched, false, nil
 }
 
+func (j *leftOuterJoiner) tryToMatchOuters(outers chunk.Iterator, inner chunk.Row, chk *chunk.Chunk, outerRowStatus []outerRowStatusFlag) (_ []outerRowStatusFlag, err error) {
+	j.chk.Reset()
+	chkForJoin := j.chk
+	if len(j.conditions) == 0 {
+		chkForJoin = chk
+	}
+
+	outer, numToAppend, outerBatchSize := outers.Current(), chk.RequiredRows()-chk.NumRows(), 0
+	for ; outer != outers.End() && numToAppend > 0; outer, numToAppend, outerBatchSize = outers.Next(), numToAppend-1, outerBatchSize+1 {
```

Comment:
Don't need to calculate `outerBatchSize` in every loop. We can calculate it when the loop is finished:
```go
outer, numToAppend := outers.Current(), chk.RequiredRows()-chk.NumRows()
for ; outer != outers.End() && numToAppend > 0; outer, numToAppend = outers.Next(), numToAppend-1 {
    j.makeJoinRowToChunk(chkForJoin, outer, inner)
}

outerBatchSize := chk.RequiredRows() - chk.NumRows() - numToAppend
// do something
```

BTW, the for loop looks a little complicated, we can simplify it to:
```go
for outer != outers.End() && numToAppend > 0 {
    j.makeJoinRowToChunk(chkForJoin, outer, inner)
    outer = outers.Next()
    numToAppend = numToAppend - 1
}
```